tests/components/vodafone_station/test_sensor.py

```diff
@@ -0,0 +1,124 @@
+"""Tests for Vodafone Station sensor platform."""
+
+from copy import deepcopy
+from datetime import datetime
+from unittest.mock import patch
+
+from freezegun.api import FrozenDateTimeFactory
+import pytest
+
+from homeassistant.components.vodafone_station.const import (
+    DOMAIN,
+    LINE_TYPES,
+    SCAN_INTERVAL,
+)
+from homeassistant.core import HomeAssistant
+
+from .const import DEVICE_DATA_QUERY, MOCK_USER_DATA, SENSOR_DATA_QUERY
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+
+@pytest.mark.parametrize(
+    ("connection_type", "index"),
+    [
+        ("dsl_ipaddr", 0),
+        ("fiber_ipaddr", 1),
+        ("vf_internet_key_ip_addr", 2),
+    ],
+)
+async def test_active_connection_type(
+    hass: HomeAssistant, freezer: FrozenDateTimeFactory, connection_type, index
+) -> None:
+    """Test device connection type."""
+
+    entry = MockConfigEntry(domain=DOMAIN, data=MOCK_USER_DATA)
+    entry.add_to_hass(hass)
+
+    with (
+        patch("aiovodafone.api.VodafoneStationSercommApi.login") as mock_login,
+        patch(
+            "aiovodafone.api.VodafoneStationSercommApi.get_devices_data",
+            return_value=DEVICE_DATA_QUERY,
+        ) as mock_devices_data,
+        patch(
+            "aiovodafone.api.VodafoneStationSercommApi.get_sensor_data",
+            return_value=SENSOR_DATA_QUERY,
+        ) as mock_sensor_data,
+    ):
```

Comment:
I would recommend to just patch out the whole object instead of doing separate functions like this. Object mocks are easier to understand and have a lower chance of leaving behind lingering tests (those happen when you do a real call to the library, which can start all kinds of processes)

Reply:
Thx a lot @joostlek for your suggestions and help !